packages/rocketchat-language-plugin/server/server.js

```diff
@@ -0,0 +1,21 @@
+Accounts.onLogin(function(user) {
+	const languages_codes = user.connection.httpHeaders['accept-language'];
+
+
+	const parser = Npm.require('accept-language-parser');
+	const languages = Npm.require('languages');
+
+	const languages_parsed = parser.parse(languages_codes);
+	const priority_language = languages_parsed[0].code;
+
+	const priority_language_fullname = languages.getLanguageInfo(priority_language);
+	const room = RocketChat.models.Rooms.findOneByIdOrName(priority_language_fullname.name);
+	if (!room) {
+		Meteor.call('createChannel', priority_language_fullname.name, []);
+
+	} else {
+
+		Meteor.call('joinRoom', room._id);
```

Comment:
We should also show them a message in UI, when they open a channel. Something like "You were added to the channel because ... " "we detected that you are coming from this country", or "because we detected that you speak this language".